src/autolabel/confidence.py

```diff
@@ -76,6 +77,47 @@ def logprob_average(
                 count += 1
         return logprob_cumulative ** (1.0 / count) if count > 0 else 0
 
+    def _logprob_average_per_label(
+        self,
+        logprobs: list,
+        label: str,
+        delimiter: str = ";",
+        **kwargs,
+    ) -> Dict[str, float]:
+        logprob_per_label = {}
+        curr_logprob_average = self.logprob_average(logprobs)
+        logprob_per_label = {
+            curr_label: curr_logprob_average for curr_label in label.split(delimiter)
+        }
+        conf_label_keys, prev_key_index, curr_key = {}, 0, ""
+        for i in range(len(logprobs)):
+            for curr_chars in list(logprobs[i].keys())[0]:
+                if delimiter in curr_chars:
+                    curr_key += curr_chars.split(delimiter)[0]
+                    conf_label_keys[curr_key] = self.logprob_average(
+                        logprobs[prev_key_index:i]
+                    )
+                    prev_key_index = i
+                    curr_key = curr_chars.split(delimiter)[-1]
+                else:
+                    curr_key += curr_chars
+        if len(curr_key) > 0:
+            conf_label_keys[curr_key] = self.logprob_average(logprobs[prev_key_index:])
+
+        for conf_label_candiate in conf_label_keys:
+            closest_match, closest_match_score = None, 0
+            for label in logprob_per_label:
+                longest_substring = difflib.SequenceMatcher(
+                    None, label, conf_label_candiate
+                ).find_longest_match(0, len(label), 0, len(conf_label_candiate))
+                if longest_substring.size > closest_match_score:
```

Comment:
Changed the metric to be proportion of the largest string that overlaps, instead of the number of characters. I think that should take care of this case.